aks/postgres/outputs.tf

```
@@ -42,3 +42,7 @@ output "azure_backup_storage_account_name" {
 output "azure_backup_storage_container_name" {
   value = local.azure_enable_backup_storage ? azurerm_storage_container.backup[0].name : null
 }
+
+output "azure_server_id" {
+  value = var.use_azure ? azurerm_postgresql_flexible_server.main[0].id : null
```

```suggestion
  value = azurerm_postgresql_flexible_server.main[0].id
```

This would be the same I think as the server will only be set if `var.use_azure` is `true`, but I'm okay with it being explicit too.

---

I assume terraform would have failed if not using azure

---

Hmm maybe you're right actually, I think I was thinking of using `try` but we're not doing that here.